### PR TITLE
fix merging top-level config values

### LINKED ISSUE #1307: Unable to update config values specified in default.toml - 0.10.1
With a service running under hab-sup 0.10.1, I am unable to set any service cfg values if there is a default specified in the package's `default.toml`. If there is no default set, the config value is set appropriately.

Steps to reproduce:
 * enter a 0.10.1 studio
 * run `HAB_REDIS="tcp-backlog = 128" hab start core/redis` and then stop after redis comes up successfully
 * Note that the backlog warning is still displayed
 * `grep tcp-backlog /hab/svc/redis/config/redis.config` and see that the value is still set to 511

- [x] The OS (including version) where you are running any of the Habitat commands.

macOS Sierra, docker 1.12.1

- [x] Debug/backtrace of the command you are trying to run. You can set the following environment variables before running the `hab` command to generate a trace:

		RUST_LOG=debug RUST_BACKTRACE=1 hab ...
```
[17][default:/hab/pkgs/core/hab-sup:0]# RUST_LOG=debug RUST_BACKTRACE=1 HAB_REDIS="tcp-backlog = 128" hab start core/redis
DEBUG:habitat_common::ui: InputStream(stdin): { is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stdout): { is_colored(): false, supports_color(): false, is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stderr): { is_colored(): false, supports_color(): false, is_a_terminal(): true }
DEBUG:hab::exec: Calling execv: ("/hab/pkgs/core/hab-sup/0.10.1/20160929220948/bin/hab-sup") ["start", "core/redis"]
DEBUG:hab_sup: clap matches ArgMatches { args: {}, subcommand: Some(SubCommand { name: "start", matches: ArgMatches { args: {"pkg_ident_or_artifact": MatchedArg { occurs: 1, vals: {1: "core/redis"} }}, subcommand: None, usage: Some("USAGE:\n    hab-sup start [FLAGS] [OPTIONS] <pkg_ident_or_artifact>") } }), usage: Some("USAGE:\n    hab-sup [FLAGS] [SUBCOMMAND]") }
DEBUG:hab_sup: subcommand name "start"
DEBUG:hab_sup: Subcommand matches ArgMatches { args: {"pkg_ident_or_artifact": MatchedArg { occurs: 1, vals: {1: "core/redis"} }}, subcommand: None, usage: Some("USAGE:\n    hab-sup start [FLAGS] [OPTIONS] <pkg_ident_or_artifact>") }
DEBUG:hab_sup: Gossip IP = 172.17.0.2
DEBUG:hab_sup: Gossip port = 9634
DEBUG:hab_sup: HTTP IP = 0.0.0.0
DEBUG:hab_sup: HTTP port = 9631
DEBUG:hab_sup: Config:
Config { command: Start, package: PackageIdent { origin: "core", name: "redis", version: None, release: None }, local_artifact: None, url: "https://willem.habitat.sh/v1/depot", topology: Standalone, group: "default", path: "/hab/svc/hab-sup/data", archive: "", bind: [], key: "", email: None, expire_days: None, gossip_listen_ip: "172.17.0.2", gossip_listen_port: 9634, http_listen_ip: "0.0.0.0", http_listen_port: 9631, userkey: None, servicekey: None, infile: None, outfile: None, gossip_peer: [], gossip_permanent: false, update_strategy: None, service_group: "", file_path: "", version_number: 0, organization: None, ring: None, config_from: None }
hab-sup(MN): Starting core/redis
DEBUG:habitat_common::ui: InputStream(stdin): { is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stdout): { is_colored(): false, supports_color(): false, is_a_terminal(): true }
DEBUG:habitat_common::ui: OutputStream(stderr): { is_colored(): false, supports_color(): false, is_a_terminal(): true }
DEBUG:habitat_sup::command::start: Setting the PATH to /hab/pkgs/core/redis/3.2.3/20160920131015/bin:/hab/pkgs/core/glibc/2.22/20160612063629/bin:/hab/pkgs/core/busybox-static/1.24.2/20160926152859/bin:/hab/bin::/hab/pkgs/core/hab-plan-build/0.10.1/20160929223059/bin:/hab/pkgs/core/diffutils/3.3/20160729193321/bin:/hab/pkgs/core/less/481/20160729193028/bin:/hab/pkgs/core/make/4.2.1/20160729193703/bin:/hab/pkgs/core/mg/20160118/20160729215442/bin:/hab/pkgs/core/util-linux/2.27.1/20160729195738/bin:/hab/pkgs/core/vim/8.0.0004/20160914213245/bin:/hab/pkgs/core/acl/2.2.52/20160612075215/bin:/hab/pkgs/core/attr/2.4.47/20160612075207/bin:/hab/pkgs/core/bash/4.3.42/20160729192720/bin:/hab/pkgs/core/binutils/2.25.1/20160612064534/bin:/hab/pkgs/core/bzip2/1.0.6/20160612075040/bin:/hab/pkgs/core/coreutils/8.25/20160729192409/bin:/hab/pkgs/core/file/5.24/20160612064523/bin:/hab/pkgs/core/findutils/4.4.2/20160729193346/bin:/hab/pkgs/core/gawk/4.1.3/20160729192843/bin:/hab/pkgs/core/glibc/2.22/20160612063629/bin:/hab/pkgs/core/grep/2.22/20160729192643/bin:/hab/pkgs/core/gzip/1.6/20160729193649/bin:/hab/pkgs/core/hab/0.10.1/20160929220055/bin:/hab/pkgs/core/libcap/2.24/20160612075226/bin:/hab/pkgs/core/libidn/1.32/20160729195459/bin:/hab/pkgs/core/ncurses/6.0/20160612075116/bin:/hab/pkgs/core/openssl/1.0.2j/20160926152543/bin:/hab/pkgs/core/pcre/8.38/20160729192620/bin:/hab/pkgs/core/sed/4.2.2/20160612075228/bin:/hab/pkgs/core/tar/1.28/20160729194917/bin:/hab/pkgs/core/unzip/6.0/20160729203724/bin:/hab/pkgs/core/wget/1.18/20160926152823/bin:/hab/pkgs/core/xz/5.2.2/20160729193408/bin
DEBUG:habitat_sup::util::users: SVC_USER = hab
DEBUG:habitat_sup::util::users: SVC_GROUP = hab
hab-sup(TP): Child process will run as user=hab, group=hab
DEBUG:habitat_sup::util::sys: Determining host name
DEBUG:habitat_sup::util::sys: Hostname = 552c68f68349
DEBUG:habitat_sup::util::sys: Determining host name
DEBUG:habitat_sup::util::sys: Hostname = 552c68f68349
hab-sup(GS): Supervisor 172.17.0.2: 03b29256-a629-4012-affa-c3e82d7554bb
hab-sup(GS): Census redis.default: c37c4029-ce3e-47c6-884f-c7ca4352c126
DEBUG:habitat_sup::gossip::rumor: Adding rumor Rumor { id: Uuid("03b29256-a629-4012-affa-c3e82d7554bb"), payload: Member(Member { id: Uuid("03b29256-a629-4012-affa-c3e82d7554bb"), hostname: "552c68f68349", ip: "172.17.0.2", gossip_listener: "172.17.0.2:9634", incarnation: LamportClock { counter: 0 }, health: Alive, permanent: false }) }
DEBUG:habitat_sup::gossip::rumor: Adding rumor Rumor { id: Uuid("c37c4029-ce3e-47c6-884f-c7ca4352c126"), payload: CensusEntry(CensusEntry { id: Uuid("c37c4029-ce3e-47c6-884f-c7ca4352c126"), member_id: Uuid("03b29256-a629-4012-affa-c3e82d7554bb"), hostname: "552c68f68349", ip: "172.17.0.2", suitability: 0, port: Some("6379"), exposes: Some(["6379"]), leader: false, follower: false, data_init: false, vote: None, election: None, needs_write: None, initialized: false, keep_me: true, service: "redis", group: "default", alive: true, suspect: false, confirmed: false, detached: false, incarnation: LamportClock { counter: 0 } }) }
hab-sup(GS): Starting inbound gossip listener
hab-sup(GS): Starting outbound gossip distributor
hab-sup(GS): Starting gossip failure detector
hab-sup(CN): Starting census health adjuster
DEBUG:habitat_sup::service_config: Failed to open user.toml: No such file or directory (os error 2)
DEBUG:habitat_sup::service_config: Failed to open gossip.toml: No such file or directory (os error 2)
DEBUG:habitat_sup::util::users: User/group not specified in package, running with default
DEBUG:habitat_sup::util::users: User/group not specified in package, running with default
DEBUG:habitat_sup::gossip::detector: Detector State: Detector {
    open_requests: {}
}
DEBUG:habitat_sup::util::users: SVC_USER = hab
DEBUG:habitat_sup::util::users: SVC_GROUP = hab
DEBUG:habitat_sup::util::sys: Determining host name
DEBUG:habitat_sup::util::sys: Hostname = 552c68f68349
DEBUG:habitat_sup::util::users: SVC_USER = hab
DEBUG:habitat_sup::util::users: SVC_GROUP = hab
DEBUG:habitat_sup::package: Creating svc paths
DEBUG:habitat_sup::package: Creating dir with subdirs: "/hab/svc/redis"
DEBUG:habitat_sup::package: Creating dir with subdirs: "/hab/svc/redis/config"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/redis/config" to "hab"
DEBUG:habitat_sup::package: Creating dir with subdirs: "/hab/svc/redis/data"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/redis/data" to "hab"
DEBUG:habitat_sup::package: Creating dir with subdirs: "/hab/svc/redis/files"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/redis/files" to "hab"
DEBUG:habitat_sup::package: Creating dir with subdirs: "/hab/svc/redis/hooks"
DEBUG:habitat_sup::package: Creating dir with subdirs: "/hab/svc/redis/var"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/redis/var" to "hab"
DEBUG:habitat_sup::package: Creating dir with subdirs: "/hab/svc/redis/static"
DEBUG:habitat_core::util::perm: Attempting to set owner of "/hab/svc/redis/static" to "hab"
DEBUG:habitat_sup::package: Creating dir with subdirs: "/hab/svc/redis/toml"
DEBUG:habitat_sup::package: Copying the run file
DEBUG:habitat_sup::package: svc_run = /hab/svc/redis/run
DEBUG:habitat_sup::package: run file = /hab/pkgs/core/redis/3.2.3/20160920131015/run
DEBUG:habitat_sup::package: svc_run file = /hab/svc/redis/run
DEBUG:hyper::server: threads = 16
DEBUG:habitat_sup::service_config: Registering handlebars helpers
DEBUG:habitat_sup::service_config: Registering configuration templates
DEBUG:habitat_sup::package: Loading configuration from "/hab/pkgs/core/redis/3.2.3/20160920131015/config"
DEBUG:habitat_sup::package: Looking in "redis.config"
DEBUG:habitat_sup::service_config: Config template redis.config from "/hab/pkgs/core/redis/3.2.3/20160920131015/config/redis.config"
DEBUG:habitat_sup::service_config: Rendering template redis.config
DEBUG:handlebars::context: visiting path: "."/"pkg.svc_var_path"
DEBUG:handlebars::context: visiting path: "."/"cfg.port"
DEBUG:handlebars::context: visiting path: "."/"cfg.tcp-backlog"
DEBUG:handlebars::context: visiting path: "."/"cfg.bind"
DEBUG:handlebars::context: visiting path: "."/"cfg.protected-mode"
DEBUG:handlebars::context: visiting path: "."/"cfg.protected-mode"
DEBUG:handlebars::context: visiting path: "."/"cfg.unixsocket"
DEBUG:handlebars::context: visiting path: "."/"cfg.unixsocketperm"
DEBUG:handlebars::context: visiting path: "."/"cfg.timeout"
DEBUG:handlebars::context: visiting path: "."/"cfg.tcp-keepalive"
DEBUG:handlebars::context: visiting path: "."/"cfg.loglevel"
DEBUG:handlebars::context: visiting path: "."/"cfg.logfile"
DEBUG:handlebars::context: visiting path: "."/"cfg.syslog-enabled"
DEBUG:handlebars::context: visiting path: "."/"cfg.syslog-ident"
DEBUG:handlebars::context: visiting path: "."/"cfg.syslog-facility"
DEBUG:handlebars::context: visiting path: "."/"cfg.databases"
DEBUG:handlebars::context: visiting path: "."/"cfg.save"
DEBUG:handlebars::helpers::helper_each: each value Array([Object({"keys": I64(1), "sec": I64(900)}), Object({"keys": I64(10), "sec": I64(300)}), Object({"keys": I64(10000), "sec": I64(60)})])
DEBUG:handlebars::helpers::helper_each: each path "./cfg.save.[0]"
DEBUG:handlebars::context: visiting path: "./cfg.save.[0]"/"sec"
DEBUG:handlebars::context: visiting path: "./cfg.save.[0]"/"keys"
DEBUG:handlebars::helpers::helper_each: each path "./cfg.save.[1]"
DEBUG:handlebars::context: visiting path: "./cfg.save.[1]"/"sec"
DEBUG:handlebars::context: visiting path: "./cfg.save.[1]"/"keys"
DEBUG:handlebars::helpers::helper_each: each path "./cfg.save.[2]"
DEBUG:handlebars::context: visiting path: "./cfg.save.[2]"/"sec"
DEBUG:handlebars::context: visiting path: "./cfg.save.[2]"/"keys"
DEBUG:handlebars::context: visiting path: "."/"cfg.stop-writes-on-bgsave-error"
DEBUG:handlebars::context: visiting path: "."/"cfg.rbcompression"
DEBUG:handlebars::context: visiting path: "."/"cfg.rbchecksum"
DEBUG:handlebars::context: visiting path: "."/"cfg.dbfilename"
DEBUG:handlebars::context: visiting path: "."/"pkg.svc_data_path"
DEBUG:handlebars::context: visiting path: "."/"svc.me.follower"
DEBUG:handlebars::context: visiting path: "."/"cfg.masterauth"
DEBUG:handlebars::context: visiting path: "."/"cfg.slave-serve-stale-data"
DEBUG:handlebars::context: visiting path: "."/"cfg.slave-read-only"
DEBUG:handlebars::context: visiting path: "."/"cfg.repl-ping-slave-period"
DEBUG:handlebars::context: visiting path: "."/"cfg.repl-timeout"
DEBUG:handlebars::context: visiting path: "."/"cfg.repl-disable-tcp-nodelay"
DEBUG:handlebars::context: visiting path: "."/"cfg.repl-backlog-size"
DEBUG:handlebars::context: visiting path: "."/"cfg.repl-backlog-size"
DEBUG:handlebars::context: visiting path: "."/"cfg.repl-backlog-ttl"
DEBUG:handlebars::context: visiting path: "."/"cfg.repl-backlog-ttl"
DEBUG:handlebars::context: visiting path: "."/"cfg.slave-priority"
DEBUG:handlebars::context: visiting path: "."/"cfg.min-slaves-to-write"
DEBUG:handlebars::context: visiting path: "."/"cfg.min-slaves-max-lag"
DEBUG:handlebars::context: visiting path: "."/"cfg.requirepass"
DEBUG:handlebars::context: visiting path: "."/"cfg.appendonly"
DEBUG:handlebars::context: visiting path: "."/"cfg.appendfsync"
DEBUG:handlebars::context: visiting path: "."/"cfg.no-appendfsync-on-rewrite"
DEBUG:habitat_sup::service_config: Configuration /hab/svc/redis/config/redis.config does not exist; restarting
hab-sup(SC): Updated redis.config
DEBUG:habitat_sup::package: Copying the run file
DEBUG:habitat_sup::package: svc_run = /hab/svc/redis/run
DEBUG:habitat_sup::package: run file = /hab/pkgs/core/redis/3.2.3/20160920131015/run
DEBUG:habitat_sup::package: svc_run file = /hab/svc/redis/run
hab-sup(TP): Restarting because the service config was updated via the census
DEBUG:habitat_sup::gossip::detector: Detector State: Detector {
    open_requests: {}
}
DEBUG:habitat_sup::gossip::detector: Detector State: Detector {
    open_requests: {}
}
redis(SV): Starting
```
	
- [x] Current Habitat environment variables where the `hab` command or supervisor is running.  These can be gathered using:

		env | grep HAB | grep -v HAB_AUTH_TOKEN

	- [x] If you're running into problems with the [Habitat Studio](https://www.habitat.sh/docs/concepts-studio/), please use the `env | grep ...` command specified above to include environment variables:
		- [x] a. before starting the studio

```
(empty)
```
		- [x] b. inside the studio

```
18][default:/hab/pkgs/core/hab-sup:0]# env | grep HAB | grep -v HAB_AUTH_TOKEN
HAB_ORIGIN=adamleff
```

### PR DESCRIPTION
**Pull request by robbkidd**
Added a test case for top-level config values not getting overriden now that values in tables are (from #1294). Actual fix is pending.

Fixes #1307

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by metadave** [comment]
note: we changed `left` & `right` to `ovr` & `dflt` to make it easier to read. And it is!

**Comment by metadave** [comment]
ping @habitat-sh/habitat-core-maintainers

**Comment by robbkidd** [comment]
Doing some more testing and maybe folding those tests into test cases.

File: components/sup/src/service_config.rs [hunk 1]
```diff
@@ -725,17 +721,21 @@ mod test {
 
 
     #[test]
-    fn merge() {
-        let default_config = "[server]
+    fn dp_merge() {
```
Comment (metadave) [inline components/sup/src/service_config.rs]: this should be renamed

**Comment by metadave** [comment]
we're still investigating this fix, will reopen when complete.

**Comment by metadave** [comment]
🍰  🍻 @robbkidd @fnichol

**Comment by metadave** [comment]
I'm not only a +1 on this PR, I'm also a client.

**Comment by robbkidd** [comment]
Appveyor, GO!
force push

**Comment by fnichol** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/164160697).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/164160697).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**